dev/plugins/hu.elte.txtuml.export.plantuml/src/hu/elte/txtuml/export/plantuml/generator/PlantUmlCompiler.java

```diff
@@ -0,0 +1,314 @@
+	 * it.
+	 * 
+	 * @param fullyQualifiedName
+	 *            the fully qualified name of the fragment we are compiling
+	 * @return
+	 */
+	public String compile(String fullyQualifiedName) {
+
+		if (this.cache.hasCompiledElement(fullyQualifiedName)) {
+			CompileCache cache = this.cache.getCompiledElement(fullyQualifiedName);
+			activeLifelines.addAll(cache.getActiveLifeines());
+			return this.compiledOutput;
+		}
+
+		boolean hasDecl = false;
+		MethodDeclaration declaration = null;
+
+		for (MethodDeclaration decl : fragments) {
+			if (PlantUmlCompiler.getFullyQualifiedName(decl).equals(fullyQualifiedName)) {
+				hasDecl = true;
+				declaration = decl;
+				break;
+			}
+		}
+
+		if (hasDecl) {
+			PlantUmlCompiler compiler = new PlantUmlCompiler(this.lifelines, this.fragments, true);
+			compiler.activeLifelines = activeLifelines;
+			declaration.accept(compiler);
+			this.cache.addCompiledElement(PlantUmlCompiler.getFullyQualifiedName(declaration),
+					compiler.getCompiledOutput(), compiler.activeLifelines);
+			activeLifelines.addAll(compiler.activeLifelines);
+			return compiler.getCompiledOutput();
+		}
+
+		return null;
+	}
+
+	/**
+	 * End of visit actions
+	 */
+	@Override
+	public void postVisit(ASTNode node) {
+		BaseSeqdiagExporter<?> expt = BaseSeqdiagExporter.createExporter(node, this);
+
+		if (!expQueue.isEmpty() && expt != null && !expt.skippedStatement(node)) {
+			BaseSeqdiagExporter<?> exp = expQueue.peek();
+
+			if (expt.getClass().isInstance(exp)) {
+				exp.endVisit(node);
+			} else if (!expt.getClass().isInstance(exp) && !expt.skippedStatement(node)) {
+				errors.add(node);
+			}
+		}
+	}
+
+	/**
+	 * Get the compilation errors
+	 * 
+	 * @return
+	 */
+	public List<String> getErrors() {
+		ArrayList<String> errList = new ArrayList<String>();
+		for (ASTNode error : errors) {
+			errList.add("Error! Couldn't parse the following statement: " + error.toString() + "\n");
+		}
+
+		if (!expQueue.isEmpty()) {
+			errList.add("There was a total of " + expQueue.size() + "half-processed or badly processed statements");
+		}
+
+		return errList;
+	}
+
+	/*
+	 * compile time statement handling
+	 */
+
+	public void preProcessedStatement(BaseSeqdiagExporter<? extends ASTNode> exporter) {
+		expQueue.push(exporter);
+	}
+
+	public void postProcessedStatement() {
+		expQueue.pop();
+	}
+
+	/**
+	 * 
+	 * checks if lifeline is active( used in exporting)
+	 * 
+	 * @param lifeline
+	 * @return true if the lifeline is activated
+	 */
+	public boolean lifelineIsActive(String lifeline) {
+		if (activeLifelines.contains(lifeline)) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * activates lifeline and does the needed supplementary operations
+	 * 
+	 * @param lifeline
+	 *            lifeline to activate
+	 */
+	public void activateLifeline(String lifeline) {
+
+		if (!lifelineIsActive(lifeline)) {
+			compiledOutput += "activate " + lifeline + "\n";
+		}
+
+		activeLifelines.add(lifeline);
```

Comment:
Shouldn't this be in the if statement?

Reply:
You are right, it should not activate an already activated lifeline.
By the way this solution works now, because this method is called only from one place, and there is a check performed for this case.